tests/PHPStan/Analyser/nsrt/bug-12107.php

```diff
@@ -0,0 +1,43 @@
+<?php declare(strict_types = 1);
+
+namespace Bug12107;
+
+use LogicException;
+use Throwable;
+
+use function PHPStan\Testing\assertType;
+
+class HelloWorld
+{
+	public function sayHello(Throwable $e1, LogicException $e2): void
+	{
+		if ($e1 instanceof $e2) {
+			return;
+		}
+
+		assertType('Throwable', $e1);
+		assertType('bool', $e1 instanceof $e2); // could be false
```

Comment:
I'm unsure to say PHPStan that
- $e1 is not `Throwable~LogicException` but still `Throwable`
- the specific expression `$e1 instanceof $e2` is false because already computed.

But I'm not sure it's a big loss.